partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -584,18 +584,21 @@ internal class PlanTyper(
 
             // Type the arguments
```

Comment:
I'm curious if https://github.com/partiql/partiql-lang-kotlin/blob/v1/partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt#L1357-L1405 can be a replacement for this? This is a lot of logic that isn't in V1. Ideally we can bring the latest to 0.14.+

Reply:
+1 to Robert's question. Can the same logic from `v1` be ported to this PR/`main`? Have a couple concerns that
- the PR may have a different typing behavior from the `v1` branch
- the code changes may make it tricky to merge `main` into `v1` (or `v1` into `main`)